prover/src/lib.rs

```diff
@@ -496,44 +480,53 @@ pub trait Prover {
-        #[cfg(feature = "std")]
-        debug!(
-            "Converted constraint evaluations into {} composition polynomial columns of degree {} in {} ms",
-            composition_poly.num_columns(),
-            composition_poly.column_degree(),
-            now.elapsed().as_millis()
-        );
+        let composition_poly = info_span!(
+            "Converted constraint evaluations into composition polynomial columns",
+            columns_number = field::Empty,
+            degree = field::Empty
+        )
+        .in_scope(|| {
+            let composition_poly =
+                CompositionPoly::new(composition_poly_trace, domain, num_trace_poly_columns);
+
+            tracing::Span::current().record("columns_number", composition_poly.num_columns());
+            tracing::Span::current().record("degree", composition_poly.column_degree());
+
+            composition_poly
+        });
 
         // then, evaluate composition polynomial columns over the LDE domain
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let composed_evaluations = RowMatrix::evaluate_polys_over::<DEFAULT_SEGMENT_WIDTH>(
-            composition_poly.data(),
-            domain,
-        );
-        #[cfg(feature = "std")]
-        debug!(
-            "Evaluated {} composition polynomial columns over LDE domain (2^{} elements) in {} ms",
-            composed_evaluations.num_cols(),
-            composed_evaluations.num_rows().ilog2(),
-            now.elapsed().as_millis()
-        );
+        let composed_evaluations = info_span!(
+            "Evaluated composition polynomial columns over LDE domain",
+            columns_number = field::Empty,
+            domain_size = field::Empty
+        )
+        .in_scope(|| {
+            let composed_evaluations = RowMatrix::evaluate_polys_over::<DEFAULT_SEGMENT_WIDTH>(
+                composition_poly.data(),
+                domain,
+            );
+
+            tracing::Span::current().record("columns_number", composed_evaluations.num_cols());
+            tracing::Span::current().record("domain_size", composed_evaluations.num_rows().ilog2());
+
+            composed_evaluations
+        });
 
         // finally, build constraint evaluation commitment
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let commitment = composed_evaluations.commit_to_rows();
-        let constraint_commitment = ConstraintCommitment::new(composed_evaluations, commitment);
-        #[cfg(feature = "std")]
-        debug!(
-            "Computed constraint evaluation commitment (Merkle tree of depth {}) in {} ms",
-            constraint_commitment.tree_depth(),
-            now.elapsed().as_millis()
-        );
+        let constraint_commitment = info_span!(
+            "Computed constraint evaluation commitment",
+            merkle_tree_depth = field::Empty
+        )
+        .in_scope(|| {
```

Comment:
Let's make the following changes here:

1. Change the name to "compute_constraint_evaluation_commitment".
2. For `tree_depth` field let's use `domain.lde_domain_size().ilog2()`.
3. Assert that `tree_depth` is indeed `lde_domain_size().ilog2()`.